### ISSUE TITLE
Weirdness with old survey and new dialogues over SMS

### ISSUE THREAD

**Issue by badlard** [praekeltfoundation/vumi-go#724]
Took the following steps:
- I set up an old survey over SMS and a dialogue over SMS
- for each conversation, I initiated the survey/dialogue by sending the first question to my MSISDN.

I had the following experiences:
- I never received the first question
- I was presented with the first question twice
- after answering the first question, I was again presented with the first question
- after answering the first question, I was presented with the second question (as it should)
- after answering the second question, I was then presented with the second and third question at once.

Even weirder:
- Eve also said that when users dialed into the survey (i.e. she didn't send it out) then the experience was fine. 
- I was able to replicate this - i.e. when I sent a SMS to the long number from my phone then questions were offered to me in the right order.

First question presented twice:
![screen shot 2013-09-30 at 1 55 18 pm](https://f.cloud.github.com/assets/1521387/1236667/eed8505e-29c6-11e3-9f97-00bc093f7b2e.png)

First question presented again after answering:
![screen shot 2013-09-30 at 1 55 06 pm](https://f.cloud.github.com/assets/1521387/1236670/f484daea-29c6-11e3-90eb-f237e63dd7d1.png)

Thank you message presented before answering the last question:
![screen shot 2013-09-30 at 1 54 59 pm](https://f.cloud.github.com/assets/1521387/1236672/fc3fea18-29c6-11e3-8607-1fb13817d217.png)

**Comment by badlard** [comment]
I am now receiving some of the first questions that weren't sent to me when I initiated the send two days ago. Thanks to #717

**Comment by hodgestar** [comment]
That would be because we just deployed the fix to #717. :)

**Comment by badlard** [comment]
Dialogue: 
- I received the first question - I responded '1' and never received a response
- after waiting 10 minutes, I initiated the survey to myself again
- I then received the second question, as well as two copies of the first question.
- I responded '1'
- I then received the third question (I guess Vumi thought I was responding to the second question?)
- I responded '1' and then received the 'thank you' message
- (It seems like the timing got messed up here when I initiated the survey to myself again...)

Dialogue - second try (or third):
- I received the first question twice
- I responded '1'
- I received the first question again
- I responded '1'
- I received the second question
- I responded '1'
- (after a delay) I received the third question
- I responded '1'
- (after a very long delay) I received the 'thank you' message

Old survey - first try:
- I received the first question twice, but after that moved through the survey normally 

Old survey - second try:
- I sent the survey to myself again in Vumi Go, but never received an SMS with the first question
- I later received the first question, but only after I initiated my dialogue conversation again (i.e. a seperate conversation to this one)
- I then moved through the survey normally

In summary, I am either:
- not receiving the first question 
- receiving the first question twice
- not receiving a question in response to an answer I submit (or it's much delayed)
- receiving the same question after responding to it
- only receiving a question once I initiate another SMS from Vumi Go

**Comment by hodgestar** [comment]
We've migrated the dialogues to sandbox version 2 which should hopefully fix this -- ready for retesting.

**Comment by hodgestar** [comment]
Closing this -- we can reopen if it re-surfaces.

**Comment by badlard** [comment]
Cool, I tested with good results.

On 9 May 2014 18:31, Simon Cross notifications@github.com wrote:

> Closed #724 https://github.com/praekelt/vumi-go/issues/724.
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/praekelt/vumi-go/issues/724#event-119615160
> .

## 

_BEN ADLARD_ | PRODUCT MANAGER

_PRAEKELTFOUNDATION_.ORG
_M_:  +27823182071  |  _S_: badlard
_T_: vumigo.tumblr.com  |  *T: *@vumiapp https://twitter.com/vumiapp
